### ISSUE TITLE
Examples: amp-inline-gallery thumbnails cannot be used with a keyboard

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31160]
## WCAG Level
Level AA

## Priority
Low

## Type
Interaction

## Pages/screens/components affected
* [amp-inline-gallery](https://amp.dev/documentation/components/amp-inline-gallery/)

## Description
Good design means supporting different modes of interaction, because there are many reasons why someone might use a keyboard instead of a mouse, and if keyboard interaction is not fully supported the website is effectively unusable.

In the [`amp-inline-gallery` pagination and thumbnails Glitch](https://nutritious-tide-hornet.glitch.me/amp-inline-gallery/pagination-and-thumbnails.html), the thumbnails in the thumbnail strip can't currently be focused/operated with the keyboard.

![The thumbnail strip for the inline gallery - none of the thumbnails receives focus](https://user-images.githubusercontent.com/54933393/99181310-8f840800-2725-11eb-9e05-e9ed12f10f74.png)

```
<div class="i-amphtml-inline-gallery-thumbnails-container i-amphtml-carousel-slotted" role="listitem"  ...>
  <div class="i-amphtml-inline-gallery-thumbnails-thumbnail">
    <div ...></div>
    <amp-img class="..." layout="fill" srcset="..."
    i-amphtml-layout="fill" style="...">
      <img decoding="async" sizes="...">
    </amp-img>
  </div>
</div>
```

## User impact
Someone with mobility problems will often use a keyboard instead of a mouse to navigate a web page. A screen reader user will also tend to use a keyboard if they cannot see the screen to use the mouse. Therefore, if the website does not properly support keyboard interaction a large number of people are effectively prevented from using it.

In this instance, keyboard users will not be able to use the thumbnail strip (though they will still be able to use the main gallery's previous/next controls to move through the gallery, but at a much slower/more tedious pace).

## Required solution
Make sure that all interactive controls can be used with a keyboard.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance

In this case, we recommend using an actual `<button>` element for each image - by default, buttons will be focusable and operable using the keyboard. Also, give each thumbnail an `alt` attribute, and give the button an understandable accessible name using `aria-label`.

```
<div class="i-amphtml-inline-gallery-thumbnails-container i-amphtml-carousel-slotted" role="listitem"  ...>
  <div class="i-amphtml-inline-gallery-thumbnails-thumbnail">
    <div ...></div>
    <button ... aria-label="Scroll to image 5">
      <amp-img class="..." layout="fill" srcset="..."
      i-amphtml-layout="fill" style="...">
        <img decoding="async" sizes="..." alt="Image 5">
      </amp-img>
    </button>
  </div>
</div>
```

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check that all interactive elements can be navigated to using the `Tab` key.
2. Check that all interactive elements can be activated using the `Enter` or `Space` keys.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website  has been tested for the same issue.
* All issues identified throughout the rest of the website  have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 2.1.1 Keyboard (Level A)](https://www.w3.org/TR/WCAG21/#keyboard)

## More information
* [WebAIM: Keyboard](https://webaim.org/techniques/keyboard/)

## Test data
Test date: November 2020
